src/main/scala/scalismo/io/MeshIO.scala

```diff
@@ -583,58 +583,64 @@ object MeshIO {
   private def readPLY(file: File): Try[Either[TriangleMesh[_3D], VertexColorMesh3D]] = {
 
     // read the ply header to find out if the ply is a textured mesh in ASCII (in which case we return a failure since VTKPLYReader Update() would crash otherwise)
-    val breader = new BufferedReader(new FileReader(file))
-    val lineIterator = Iterator.continually(breader.readLine())
 
-    val headerLines = lineIterator.dropWhile(_ != "ply").takeWhile(_ != "end_header").toIndexedSeq
-
-    if (headerLines.exists(_.contains("TextureFile")) && headerLines.exists(_.contains("format ascii"))) {
-      Failure(
-        new Exception(
-          "PLY file seems to be a textured mesh in ASCII format which creates issues with the VTK ply reader. Please convert it to a binary ply or to a vertex color or shape only ply."
-        )
-      )
+    if (!file.exists()) {
+      val filename = file.getCanonicalFile
+      Failure(new IOException(s"Could not read ply file with name $filename. Reason: The file does not exist"))
     } else {
+      val breader = new BufferedReader(new FileReader(file))
+      val lineIterator = Iterator.continually(breader.readLine())
 
-      val plyReader = new vtkPLYReader()
-      plyReader.SetFileName(file.getAbsolutePath)
-
-      plyReader.Update()
+      val headerLines = lineIterator.dropWhile(_ != "ply").takeWhile(_ != "end_header").toIndexedSeq
 
-      val errCode = plyReader.GetErrorCode()
-      if (errCode != 0) {
-        return Failure(new IOException(s"Could not read ply mesh (received VTK error code $errCode"))
+      if (headerLines.exists(_.contains("TextureFile")) && headerLines.exists(_.contains("format ascii"))) {
+        Failure(
+          new IOException(
+            "PLY file seems to be a textured mesh in ASCII format which creates issues with the VTK ply reader. Please convert it to a binary ply or to a vertex color or shape only ply."
```

Comment:
Add filename to the error message as for the missing file case?

Reply:
Thanks for the review. This is a good point. I have addressed it in the last commit.